lib/vnet/osconfig_darwin.go

```diff
@@ -140,3 +140,35 @@ func vnetManagedResolverFiles() (map[string]struct{}, error) {
 	}
 	return matchingFiles, nil
 }
+
+// doWithDroppedRootPrivileges drops the privileges of the current process to those of the client
+// process that called the VNet daemon.
+func (c *osConfigurator) doWithDroppedRootPrivileges(ctx context.Context, fn func() error) (err error) {
+	rootEgid := os.Getegid()
+	rootEuid := os.Geteuid()
+
+	if err := syscall.Setegid(c.daemonClientCred.Egid); err != nil {
+		return trace.Wrap(err, "setting egid")
+	}
+	defer func() {
+		syscallErr := trace.Wrap(syscall.Setegid(rootEgid), "reverting egid")
+		err = trace.NewAggregate(err, syscallErr)
```

Comment:
`trace.aggregate` composes really poorly with itself, I'd find some other way to keep track of the multierror before returning if we really care about that.

Reply:
Solved by removing `trace.NewAggregate` altogether in favor of using `panic`.